Make NPM scripts AWESOME :)

mattma opened the pull request:
1. Bump up package dependencies
2. Restructure the existing **NPM scripts**, following the semantic of `name:subname` styles. 
3. Using `./node_modules/.bin/` path instead of **global** path. For example, if developer does not have `tslint` installed global, it won't be able to run linting. Using local copy of the binary should be able to fix this problem
4. Add a new `npm run clean` to clean up the project. Add `postinstall` script to run `typings` right after `npm install`.

mattma committed:
lint free. fix linting bugs

mattma committed:
Merge remote-tracking branch 'upstream/master'

* upstream/master:
  1.5.0
  chore(upgrade): move to @angular/core (#103)
  lint free. fix linting bugs (#100)

mattma committed:
bump up package dependencies and make npm scripts awesome

kylecordes commented on package.json:
```diff
@@ -4,14 +4,15 @@
-    "_prepublish": "npm run typings && npm run build_dist",
-    "lint": "tslint src/**.ts",
-    "typings": "typings install"
+    "clean": "./node_modules/.bin/rimraf node_modules typings dist tmp npm-debug.log && npm cache clean",
```
All of these are unnecessary. When running scripts, NPM automatically puts ./node_modules/.bin/ in the "path" so that locally installed tools will be found there. The tools do not need to be installed globally and you do not need to say ./node_modules/.bin/ in scripts.

mattma replied:
@kylecordes Thanks. I have removed prefix.

MikeRyanDev committed:
refactor(): Refactor core to remove middleware and simplify internals (#113)

BREAKING CHANGE:
  Middleware is no longer supported. There are no plans to re-implement this feature.

BREAKING CHANGE:
  `getState()`, `getValue()`, and `value` have been removed from `Store`. Instead, subscribe will deliver the last value synchronously.

MikeRyanDev committed:
2.0.0

btroncone committed:
Update README.md (#130)

Added link to new @ngrx example app

robwormald commented:
Could you squash your commits please, and I'll merge this in 👍

MikeRyanDev commented:
> Add postinstall script to run typings right after npm install.

We actually don't want this. `postinstall` will also after consumers install `@ngrx/store` from `npm`. Would probably be best as a prebuild script.

mattma commented:
@MikeRyan52 I have moved typings installation from `postinstall` to `prebuild`. It is a good suggestion. 

@robwormald I have squashed the commits into one. But here is the thing, since my master branch is merged with latest `2.0.0` codebase, [my commits history is like this](https://github.com/mattma/store/commits/master). I hope it does not alter anyone else's commits in this case. 

After I looked at commit history in depth, I think I probably messed up here. [So I started a new PR in a clean state and ready to be merged. ](https://github.com/ngrx/store/pull/142)
